### PR TITLE
Replaces single.php and page.php with singular.php

### PR DESCRIPTION
**Pull request by mathetos**
This pull request replaces single.php and page.php with singular.php with conditional logic. The result is a DRY-er file with less code and file size. This also helps highlight the power of the new singular.php template that WordPress 4.3 introduced to the Template Heirarchy.

### CODE REVIEW

**Comment by vielmetti** [comment]
The Travis build failed because of what I hope were temporary network issues.

**Comment by philiparthurmoore** [comment]
This has been happening more frequently during the last several days. Very annoying. Will take a look tomorrow to see what we can do about having a fallback in place or a way of getting around that.

@westonruter Have you run into connectivity issues much with this?

**Comment by mathetos** [comment]
Anything I need to do with regards to this PR? I believe this new file should pass all checks.

**Comment by westonruter** [comment]
> Have you run into connectivity issues much with this?

@philiparthurmoore humm, no.

File: singular.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * The template for displaying all pages, single posts and attachments
+ *
+ * This is a new template file that WordPress introduced in
+ * version 4.2. Note that it uses conditional logic to display
```
Comment (iamtakashi) [inline singular.php:6]: This template was introduced in 4.3, wasn't it? Although it won't give us a fatal error, we probably should stop the theme to be activated below 4.3 to make sure the theme works as how it's supposed to be.

Comment (mathetos) [inline singular.php:6]: Yes, you're right. Good point. The current back-compat.php enforces 4.2. Do you want to enforce 4.3+ only? If not, we'd have to do some fancy workarounds to load a separate single.php or page.php. Or include them as-is which would make singular.php moot.

Comment (iamtakashi) [inline singular.php:6]: I don't see why not to enforce 4.3+ only.

Comment (mathetos) [inline singular.php:6]: OK, so considering Travis didn't run correctly on this PR and this typo I'll resubmit. Should I update the minimum requirement in back-compat.php with my PR as well? Seems prudent.

Comment (iamtakashi) [inline singular.php:6]: It can be separate.

Comment (chipbennett) [inline singular.php:6]: While I agree wholeheartedly regarding not facilitating use of older core
versions, having a file named "singular.php" will not cause any issues with
pre-4.3 versions of WordPress. The file will simply be ignored by the
template hierarchy. Both page.php and single.php will fall back to
index.php.

Comment (mathetos) [inline singular.php:6]: Another good point. Twenty Sixteen already requires 4.2, so maybe I'll test on 4.2 to make sure index.php works fine as a fallback for single.php and page.php

Comment (iamtakashi) [inline singular.php:6]: Yes, that's correct that's why it won't give us a fatal. We all agree a new core theme shouldn't be facilitating use of older core versions, and making the theme work as expected in 4.2 makes index.php a lot bigger and that's going to be a cruft in my opinion. We can debate in a separate thread if we want but for now, we should just add singular.php and remove page.php and single.php, and leave index.php as it is.

Comment (mathetos) [inline singular.php:6]: OK, sounds good.

**Comment by karmatosed** [comment]
I think (really hope) I'm right, that https://github.com/WordPress/twentysixteen/pull/199 is a refresh of this. So closing this in light of that.

**Pull request closed without merging**